pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -105,11 +98,25 @@ func validUUIDv4(s string) bool {
 	return true
 }
 
-func validateFlavor(flavor Flavor, req flavorRequirements, fldPath *field.Path) field.ErrorList {
+// validate flavor checks to make sure that a given flavor exists and meets the minimum requrement to run a cluster
+// this function does not validate proper install config usage
+// NOTE: Because we append the result of this function to an error list, it cannot be nil
+func validateFlavor(flavorName string, ci *CloudInfo, req flavorRequirements, fldPath *field.Path) field.ErrorList {
+	if flavorName == "" {
+		return field.ErrorList{}
+	}
```

Comment:
The note is not accurate. It is perfectly fine to unpack a nil slice.
```suggestion
func validateFlavor(flavorName string, ci *CloudInfo, req flavorRequirements, fldPath *field.Path) field.ErrorList {
	if flavorName == "" {
		return nil
	}
```

Reply:
ack, will fix